### DISCUSSION TITLE
Dynamically render meta tags from each page based on the JSON data

### DISCUSSION THREAD

**Discussion by jamestagal** [plentico/plenti#312] [Q&A] [answered]
Hello all,

I wanted to understand the best way to dynamically render meta tags for my pages on my sites based on the JSON data.
Firstly I see some websites have a different ```<head>``` tag: one for each page...or dynamically include all meta data relating to that page within each.
However, in Plenti using Svelte we only have one <head> for the whole site which is in the ```head.svelte``` file. So how can I include all the meta data from each and every page correctly?

For instance, I have an ```about.json``` file with the following meta data.
```json
	"meta": {
		"title": "About Inspired Fencing",
		"desc": "Inspired Fencing is a family owned and operated fencing contractor built on workmanship, reliability, honesty, professionality & integrity based in Somerville on Mornington Peninsula.",
		"keywords": "fencing, workmanship, reliability, honesty, family owned, Mornington Peninsula, Somerville, Inspired Fencing"
	}
```
In the ```head.svelte``` file would I add them as such?

```html
<script>
    export let meta;
</script>

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <!-- Title -->
    <title>{meta.title}</title>
    <!-- Description -->
    <meta name="description" content="{meta.desc}">
    <!-- Keywords -->
    <meta name="keywords" content="{meta.keywords}">
</head>
```
But the only problem with this is that that ```export let meta```  property doesn't reference that specific JSON file. I'm just not sure how to do it. 

Any tips and suggestions are most welcome.

Regards,
Ben

**Comment by jamestagal** [comment]
I have implemented the following:

In the ```<head>``` tag of the ```head.svelte``` file, I've added the values of ```title```, ```desc```, and ```keywords``` into the content attribute of the <meta> tag:
```html
<meta name="title" content="{title}" />
<meta name="description" content="{desc}" />
<meta name="keywords" content="{keywords}" />
```
Then exported the variables, title, desc, and keywords.
```svelte
<script context="module">
  export let env, title, desc, keywords;
</script>
```
Then hopefully any ```meta``` tags in the JSON page files will be inserted for each one that has be defined.

```json
	"meta": {
		"title": "About Plentify Plumbing",
		"desc": "Plentfy Plumbing is your reliable plumbing service in Canberra ACT. Dedicated to quality, customer service, and professional expertise in all plumbing matters. We are available 24/7.",
		"keywords": "Plentify Plumbing, Canberra ACT, Plumbing, Plumbing Services, Plumber, Plumber in Canberra"
	}
```


When looking at the page source I see the following line which looks like it is getting the title as ```About``` but no description or keywords. Also the custom title in the JSON isn't being past through:

```html
<meta name=title content="About"><meta name=description><meta name=keywords>
````
Would this be the correct way to pass the meta data into the ```<head>``` in Plenti?

BTW. I also looked at JSON-LD and defining these properties in an separate file - ```seo.svelte```
but not sure how to structure the code for this to work. Here is what I was playing around with.
(This was based off a video I watched from Webjeda but he was using Sveltekit)
```svelte
<script>
    export let type, title, description;
    $: url = "$page.url.href";
</script>
<svelte:head>
    {@html `<script type="application/ld+json">
        {
            "@context": "http://schema.org",
            "@type": "${type}",
            "name": "${title}",
            "description": "${description}",
            "url": "${url}",
        }
    </script>`}
</svelte:head>
```

Again any help would be greatly appreciated.
Regards,
Ben

**Comment by jamestagal** [comment]
Hi @jimafisk 

I altered the props, see below, based on our conversation this morning to get the ```meta``` info into the ```<head>```  as follows but I still don't see any values appearing. (besides the title which has the fallback)

```svelte
 <meta name="title" content="{content?.meta?.title ?? 'Plentify Plumbing'}" />
  <meta name="description" content="{content?.meta?.desc}" />
  <meta name="keywords" content="{content?.meta?.keywords}" />
```

Looking again at the ```json``` structure the ```meta``` field sits outside of the ```components``` array [] see below, so I wonder whether   Plenti's ```{content}``` prop isn't being scoped here?

```json
{
	"components": [
		{
			"name": "cta_1072",
			"fields": {
				"subtitle": "RESOLVE YOUR PLUMBING NEEDS",
				"title": "Quickly schedule your service or contact us for prompt support!",
				"button1": {
					"text": "Schedule Our Services",
					"link": "contact"
				},
				"button2": {
					"text": "Contact Us",
					"link": "contact"
				},
				"bgImage": {
					"src": "media/images/copper-pipes.jpg",
					"alt": "copper piping",
					"width": "1280",
					"height": "568"
				}
			}
		}
	],
	"meta": {
		"title": "About Plentify Plumbing",
		"desc": "Plentfy Plumbing is your reliable plumbing service in Canberra ACT. Dedicated to quality, customer service, and professional expertise in all plumbing matters. We are available 24/7.",
		"keywords": "Plentify Plumbing, Canberra ACT, Plumbing, Plumbing Services, Plumber, Plumber in Canberra"
	}
}

```

Please suggest a fix if possible. 
Regards,
Ben

**Reply by jimafisk** [reply, marked as answer]
It looks like `content?.meta?.title` actually needs to be `content?.fields?.meta?.title` (same for `content?.fields?.meta?.desc` and `content?.fields?.meta?.keywords`). Does that work?

**Reply by jamestagal** [reply]
Yes perfect! That worked. Thanks again Jim.

Much appreciated. 😸
